provider/agent_test.go

```diff
@@ -211,6 +211,302 @@ func TestAgent_Metadata(t *testing.T) {
+					require.Len(t, state.Modules[0].Resources, 1)
+
+					resource := state.Modules[0].Resources["coder_agent.dev"]
+					require.NotNil(t, resource)
+
+					attr := resource.Primary.Attributes
+					require.Equal(t, "1", attr["resources_monitoring.#"])
+					require.Equal(t, "1", attr["resources_monitoring.0.memory.#"])
+					require.Equal(t, "2", attr["resources_monitoring.0.volume.#"])
+					require.Equal(t, "80", attr["resources_monitoring.0.memory.0.threshold"])
+					require.Equal(t, "/volume1", attr["resources_monitoring.0.volume.0.path"])
+					require.Equal(t, "100", attr["resources_monitoring.0.volume.1.threshold"])
+					require.Equal(t, "/volume2", attr["resources_monitoring.0.volume.1.path"])
+					return nil
+				},
+			}},
+		})
+	})
+
+	t.Run("OnlyMemory", func(t *testing.T) {
+		resource.Test(t, resource.TestCase{
+			ProviderFactories: coderFactory(),
+			IsUnitTest:        true,
+			Steps: []resource.TestStep{{
+				Config: `
+					provider "coder" {
+						url = "https://example.com"
+					}
+					resource "coder_agent" "dev" {
+						os = "linux"
+						arch = "amd64"
+						resources_monitoring {
+							memory {
+								enabled = true
+								threshold = 80
+							}
+						}
+					}`,
+				Check: func(state *terraform.State) error {
+					require.Len(t, state.Modules, 1)
+					require.Len(t, state.Modules[0].Resources, 1)
+
+					resource := state.Modules[0].Resources["coder_agent.dev"]
+					require.NotNil(t, resource)
+
+					attr := resource.Primary.Attributes
+					require.Equal(t, "1", attr["resources_monitoring.#"])
+					require.Equal(t, "1", attr["resources_monitoring.0.memory.#"])
+					require.Equal(t, "80", attr["resources_monitoring.0.memory.0.threshold"])
+					return nil
+				},
+			}},
+		})
+	})
+	t.Run("MultipleMemory", func(t *testing.T) {
+		resource.Test(t, resource.TestCase{
+			ProviderFactories: coderFactory(),
+			IsUnitTest:        true,
+			Steps: []resource.TestStep{{
+				Config: `
+					provider "coder" {
+						url = "https://example.com"
+					}
+					resource "coder_agent" "dev" {
+						os = "linux"
+						arch = "amd64"
+						resources_monitoring {
+							memory {
+								enabled = true
+								threshold = 80
+							}
+							memory {
+								enabled = true
+								threshold = 90
+							}
+						}
+					}`,
+				ExpectError: regexp.MustCompile(`No more than 1 "memory" blocks are allowed`),
+			}},
+		})
+	})
+
+	t.Run("InvalidThreshold", func(t *testing.T) {
+		resource.Test(t, resource.TestCase{
+			ProviderFactories: coderFactory(),
+			IsUnitTest:        true,
+			Steps: []resource.TestStep{{
+				Config: `
+					provider "coder" {
+						url = "https://example.com"
+					}
+					resource "coder_agent" "dev" {
+						os = "linux"
+						arch = "amd64"
+						resources_monitoring {
+							memory {
+								enabled = true
+								threshold = 101
+							}
+						}
+					}`,
+				Check:       nil,
+				ExpectError: regexp.MustCompile("Error running pre-apply refresh"),
```

Comment:
Huh? This doesn't seem like the validation error I would expect.

Reply:
My bad, used this one as I did not found a way to make the regexp work, a bit more work and it is now good - should be more explicit 👌